.github/workflows/sphinx.yml

```diff
@@ -0,0 +1,67 @@
+name: sphinx-build
+
+on:
+  push:
+    branches: [ master ]
+
+env:
+  BUILD_TYPE: Release
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GITHUB_ACTOR: DataMedSci
+  GITHUB_REPOSITORY: DataMedSci/gi-yagit
+
+
+jobs:
+
+  build_sphinx_job:
+    runs-on: ubuntu-latest
+    container: debian:buster-slim
+
+    steps:
+
+      - name: Get prerequisites and clone repository
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          apt-get update
+          apt-get install -y git
+          git clone "https://token:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git" .
+        shell: bash
+
+      - name: Run build script for Sphinx pages
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          apt-get update
+          apt-get install -y git rsync python3-sphinx python3-breathe python3-pip graphviz
+          pip3 install sphinx-book-theme==0.1.7
+          pip3 install breathe==4.31.0
+          cd docs/
+          make clean
+          mkdir build
+          cd source/
+          doxygen
+          cd ../
+          make html
+          git config --global user.name "${GITHUB_ACTOR}"
+          git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+          docroot=`mktemp -d`
+          rsync -av "build/html/" "${docroot}/"
+          
+          pushd "${docroot}"
+          
+          git init
```

Comment:
Most of this code can be replaced with this action
https://github.com/APTG/libdedx/blob/master/.github/workflows/docs.yml#L56